### DISCUSSION TITLE
How do I specify a shortcode for a metadata key with a dash?

### DISCUSSION THREAD

**Discussion by tgt2** [quarto-dev/quarto-cli#7711] [Q&A]
### Description

I imagine this must have an elementary solution, but for the life of me I cannot find it. I want to be able to call the value of a meta key while in the rmarkdown/latex environment, as in:

{{< meta number-sections >}}

This seems to work fine for any key without a dash in its name, but for those where there is one a warning is issued and no value returned. Is there perhaps another encoding such that the dash will be read and recognized? 

Thank you!

**Comment by mcanouil** [comment]
I don't understand your issue here, the shortcode (in 1.3 or 1.4 pre-release) has no issue with dashes in the key.

Also why are you talking about "rmarkdown" and "latex" here? This is Quarto not rmarkdown and the shortcode works the same in every formats.

Please share a small reproducible example.

````qmd
---
title: "Reproducible Quarto Document"
format: html
my-data: value
---

{{< meta my-data >}}
````

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/35b04647-a7af-4ecf-bdf7-f890b3c34aa3">

**Comment by tgt2** [comment]
Thank you!

You are right, I was mistaken about the dash. I initially noticed this behavior with 'number-sections' and it was reproduced with 'highlight-style' but not for 'title,' 'abstact,' etc., so I began to suspect the dash. 

However I see now that it also occurs with some other keys including 'toc.'

So I suppose I am now wondering how to distinguish between the keys that will have this behavior and those that will not. 

Here is an example expanding from a section of the Quarto tutorial.

https://gist.github.com/tgt2/5db1ec89a8006891c910a989d27e3491

<img width="753" alt="Screenshot 2023-11-26 at 7 07 35 PM" src="https://github.com/quarto-dev/quarto-cli/assets/16780561/075b3200-da10-4bbb-aec3-bfc2c5c0bc02">

**Reply by mcanouil** [reply]
You need to provide the "full path" to key.

Note that you can share simple document as code blocks which is much simpler and easier, whilst also allowing users to see it without having to quit the current page.

```qmd
---
title: "Quarto Document"
author: "Norah Jones"
abstract: an abstract 
my-data: value1
highlight-style: pygments
format: 
  pdf: 
    lof: true
    lot: true
    toc: true
    number-sections: true
---

{{< meta my-data >}}
{{< meta title >}}
{{< meta author >}}
{{< meta format.pdf.toc >}}
{{< meta format.pdf.lot >}}
{{< meta format.pdf.lof >}}
{{< meta format.pdf.number-sections >}}
{{< meta highlight-style >}}
```

Edit: hum after trying out, I think the "issue" (possibly designed) is that those not working aren't really metadata but actual options/variables, i.e., "Pandoc metadata" (<https://quarto.org/docs/authoring/variables.html#meta>).

What's the use case to want to display these variables/options?

**Reply by tgt2** [reply]
Thank you and Happy New Year! 

Thank you for the response. Yes, the fact that these are 'options' is the genesis of this query. It is not (in this case) for display but rather to dictate behavior: I want to modify document formatting on the basis of user input (for instance, specify the way figures should be numbered on the basis of the value of number-sections; I am using a latex preamble to do this). Obviously one can use custom metadata fields to have this functionality but it seemed more efficient to use the shortcodes already provided.

**Reply by mcanouil** [reply]
I would personally use Quarto from the command line using metadata-file to dictate internal behaviour here (*e.g.*, <https://github.com/mcanouil/demo-generate-quarto-invoices>).

You might be interested in <https://quarto.org/docs/authoring/conditional.html#matching-against-metadata>.
